testng-core/src/main/java/org/testng/TestNG.java

```diff
@@ -1083,10 +1083,15 @@ private void runSuiteAlterationListeners() {
   }
 
   private void runExecutionListeners(boolean start) {
-    for (IExecutionListener l : m_configuration.getExecutionListeners()) {
-      if (start) {
+    List<IExecutionListener> executionListeners = m_configuration.getExecutionListeners();
+    if (start) {
+      for (IExecutionListener l : executionListeners) {
         l.onExecutionStart();
-      } else {
+      }
+    } else {
+      List<IExecutionListener> executionListenersReversed = new ArrayList<>(executionListeners);
+      Collections.reverse(executionListenersReversed);
+      for (IExecutionListener l : executionListenersReversed) {
```

Comment:
Here as well, we can use the newly added method `Lists.newReversedArrayList()`

Reply:
Fixed